sdk/go/src/sawtooth_sdk/processor/processor.go

```diff
@@ -23,72 +29,76 @@ func NewTransactionProcessor(url string) *TransactionProcessor {
 	}
 }
 
+// Start connects the TransactionProcessor to a validator and starts listening
+// for requests and routing them to an appropriate handler.
 func (self *TransactionProcessor) Start() {
 	// 1. Connect and register with the validator
 	err := self.stream.Connect(self.url)
 	if err != nil {
-		fmt.Println("Failed to start:", err)
+		LOGGER.Error("Failed to start: ", err)
 		return
 	}
 	defer self.stream.Close()
 
 	err = self.register()
 	if err != nil {
-		fmt.Println("Failed to register:", err)
+		LOGGER.Error("Failed to register: ", err)
 		return
 	}
 
 	// 2. Wait for messages
 	for {
 		msg, err := self.stream.Receive()
 		if err != nil {
-			// TODO: Handle timeout
-			fmt.Println("Failed to received a message")
+			LOGGER.Error("Failed to receive a message: ", err)
 			break
 		}
 
 		if msg.MessageType != validator_pb2.Message_TP_PROCESS_REQUEST {
-			fmt.Println("Received unexpected message:", msg)
+			LOGGER.Error("Received unexpected message: ", msg)
 			break
 		}
 
 		request := &processor_pb2.TpProcessRequest{}
 		err = proto.Unmarshal(msg.Content, request)
 		if err != nil {
-			fmt.Println("Failed to unmarshal:", err)
+			LOGGER.Error("Failed to unmarshal: ", err)
 			break
 		}
 
 		header := &transaction_pb2.TransactionHeader{}
 		err = proto.Unmarshal(request.Header, header)
 		if err != nil {
-			fmt.Println("Failed to unmarshal:", err)
+			LOGGER.Error("Failed to unmarshal: ", err)
 			break
 		}
 
-		handler, err := self.FindHandler(header)
+		// 3. Try to find handler
+		handler, err := self.findHandler(header)
 		if err != nil {
-			fmt.Println(err)
+			LOGGER.Error(err)
 			break
 		}
 
+		// 4. Construct new State instance for the handler
 		contextId := request.GetContextId()
 		state := NewState(self.stream, contextId)
 
 		err = handler.Apply(request, state)
 
+		// 5. Process the handler response
 		response := &processor_pb2.TpProcessResponse{}
 		if err != nil {
 			switch e := err.(type) {
 			case *InvalidTransactionError:
-				fmt.Println(e)
+				LOGGER.Warn(e)
 				response.Status =
 					processor_pb2.TpProcessResponse_INVALID_TRANSACTION
 			case *InternalError:
-				fmt.Println(e)
+				LOGGER.Warn(e)
 				response.Status = processor_pb2.TpProcessResponse_INTERNAL_ERROR
 			default:
-				fmt.Println("Unknown error")
+				LOGGER.Warn("Unknown error")
```

Comment:
This should also be logged as an error, since the developer will probably want to do something about it.